gammapy/maps/hpxmap.py

```diff
@@ -111,29 +111,57 @@ def from_hdulist(cls, hdulist, hdu=None, hdu_bands=None):
             Map object
         """
         if hdu is None:
-            hdu = find_bintable_hdu(hdulist)
+            hdu_out = find_bintable_hdu(hdu_list)
         else:
-            hdu = hdulist[hdu]
+            hdu_out = hdu_list[hdu]
 
         if hdu_bands is None:
-            hdu_bands = find_bands_hdu(hdulist, hdu)
+            hdu_bands = find_bands_hdu(hdu_list, hdu_out)
 
+        hdu_bands_out = None
         if hdu_bands is not None:
-            hdu_bands = hdulist[hdu_bands]
+            hdu_bands_out = hdu_list[hdu_bands]
 
-        return cls.from_hdu(hdu, hdu_bands)
+        return cls.from_hdu(hdu_out, hdu_bands_out)
 
-    def to_hdulist(self, **kwargs):
-        hdu = kwargs.get('hdu', 'SKYMAP')
-        hdu_bands = kwargs.get('hdu_bands', 'BANDS')
-        hdu_out = self.make_hdu(**kwargs)
+    def to_hdulist(self, hdu='SKYMAP', hdu_bands=None, sparse=False, conv=None):
+        """Make a FITS `` with input data.
+
+        Parameters
+        ----------
+        hdu : str
+            The HDU extension name.
+        hdu_bands : str
+            The HDU extension name for BANDS table.
+        sparse : bool
+            Set INDXSCHM to SPARSE and sparsify the map by only
+            writing pixels with non-zero amplitude.
+        conv : {'fgst-ccube','fgst-template','gadf',None}, optional        
+            FITS format convention.  If None this will be set to the
+            default convention of the map.
+
+        Returns
+        -------
+        hdu_list : `~astropy.io.fits.HDUList`
+        """
+
+        hdu_bands_out = None
```

Comment:
A bit easier to read if you move that line to the else clause, no?

Reply:
Agreed.